Add an option to not install an atexit handler. An atexit() migh hurt if a ros
plugin is dlopen() and then dlclose() before main program exit.

Rejected in https://github.com/ros/ros_comm/issues/164

--- clients/roscpp/src/libros/init.cpp~	2018-08-06 19:37:33.000000000 +0200
+++ clients/roscpp/src/libros/init.cpp	2019-05-08 03:41:07.691673675 +0200
@@ -437,7 +437,8 @@
   if (!g_atexit_registered)
   {
     g_atexit_registered = true;
-    atexit(atexitCallback);
+    if (!(options & ros_init_options_NoAtExitHandler))
+      atexit(atexitCallback);
   }
 
   if (!g_global_queue)
--- clients/roscpp/include/ros/init.h~	2018-08-06 19:37:33.000000000 +0200
+++ clients/roscpp/include/ros/init.h	2019-05-08 03:40:42.164719125 +0200
@@ -62,6 +62,9 @@
    */
   NoRosout = 1 << 2,
 };
+
+/* Don't install a atexit handler */
+#define ros_init_options_NoAtExitHandler (1 << 7)
 }
 typedef init_options::InitOption InitOption;
 
